src/frontend/src/handler/create_table_as.rs

```diff
@@ -0,0 +1,266 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use futures::StreamExt;
+use itertools::Itertools;
+use pgwire::pg_response::{PgResponse, StatementType};
+use risingwave_common::catalog::{ColumnDesc, Field};
+use risingwave_common::error::{ErrorCode, Result, RwError};
+use risingwave_common::session_config::QueryMode;
+use risingwave_pb::catalog::source::Info;
+use risingwave_pb::catalog::{
+    ColumnIndex as ProstColumnIndex, Source as ProstSource, Table as ProstTable, TableSourceInfo,
+};
+use risingwave_sqlparser::ast::{ObjectName, Query, Statement};
+
+use super::create_source::make_prost_source;
+use super::create_table::{bind_sql_table_constraints, gen_materialize_plan};
+use super::{HandlerArgs, RwPgResponse};
+use crate::binder::{BoundSetExpr, BoundStatement};
+use crate::handler::query::{
+    distribute_execute, flush_for_write, gen_batch_query_plan, local_execute,
+};
+use crate::handler::util::DataChunkToRowSetAdapter;
+use crate::optimizer::PlanRef;
+use crate::scheduler::BatchPlanFragmenter;
+use crate::session::SessionImpl;
+use crate::stream_fragmenter::build_graph;
+use crate::{Binder, OptimizerContext, OptimizerContextRef, PgResponseStream};
+
+pub async fn handle_insert(handler_args: HandlerArgs, stmt: Statement) -> Result<RwPgResponse> {
```

Comment:
It seems that this is duplicated from `handle_query` mod, why not call that function directly?

Reply:
handle_insert remove unnecessary step in handle_query, such as record metrics. Whether worth it or we can directly reuse handle_query?🤔

Reply:
Hmm, I think maintaining such duplicated code is not good practice and it's not big problem with some metrics enabled, since it will not happen frequently.